probe/kubernetes/controls.go

```diff
@@ -78,33 +78,21 @@ func (r *Reporter) CapturePod(f func(xfer.Request, string, string) xfer.Response
 func (r *Reporter) CaptureResource(f func(xfer.Request, string, string, string) xfer.Response) func(xfer.Request) xfer.Response {
 	return func(req xfer.Request) xfer.Response {
 		var resource, uid string
-		for _, parser := range []struct {
-			res string
-			f   func(string) (string, bool)
-		}{
-			{report.Deployment, report.ParseDeploymentNodeID},
-		} {
-			if u, ok := parser.f(req.NodeID); ok {
-				resource, uid = parser.res, u
-				break
-			}
+		if u, ok := report.ParseDeploymentNodeID(req.NodeID); ok {
+			resource, uid = report.Deployment, u
```

Comment:
`resource` isn't really used anywhere - the one is in an error message, where it can simply be replaced with "Deployment".